src/test/java/com/thebois/models/world/generation/noises/PerlinNoiseTests.java

```diff
@@ -0,0 +1,217 @@
+package com.thebois.models.world.generation.noises;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class PerlinNoiseTests {
+
+    public static Stream<Arguments> getPositions() {
+        return Stream.of(
+            Arguments.of(1.0f, 1.0f),
+            Arguments.of(1.0f, 1.23f),
+            Arguments.of(4, 4.5f),
+            Arguments.of(0, 0),
+            Arguments.of(-1.0f, -123.1253f),
+            Arguments.of(20.1235f, -7.534f),
+            Arguments.of(0, 1.23f),
+            Arguments.of(-1.5615f, 0)
+
+        );
+    }
+
+    public static Stream<Arguments> getTwoDifferentPositions() {
+        return Stream.of(
+            Arguments.of(1.0f, 1.0f, 1.0f, 0),
+            Arguments.of(1.0f, 1.23f, -5.0f, 2.23f),
+            Arguments.of(4, 4.5f, 0, 4.5f),
+            Arguments.of(0, 0, 1, -1),
+            Arguments.of(-1.0f, -123.1253f, 1.4f, 155),
+            Arguments.of(20.1235f, -7.534f, 1, 1),
+            Arguments.of(0, 1.23f, 1452, 0),
+            Arguments.of(-1.5615f, 0, 0.0523f, -0.5662f)
+
+        );
+    }
+
+    public static Stream<Arguments> getPerlinNoiseSettingsParameters() {
+        return Stream.of(
+            Arguments.of(1, 1f, 1f, 1f),
+            Arguments.of(1, 1.23f, 1.234f, 0.23423f),
+            Arguments.of(4, 4.5f, 0.001f, 0.9f),
+            Arguments.of(4, 4.5f, 0.001f, 0));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseSameResultWithSameValues(
+        final float posX, final float posY) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 1, 1, 1);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise.sample(posX, posY);
+        heightValue2 = perlinNoise.sample(posX, posY);
+
+        // Assert
+        assertThat(heightValue1).isEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPerlinNoiseSettingsParameters")
+    public void perlinNoiseSameResultWithSameValues(
```

Comment:
style

You've got two tests with the same name, this one should probably be called `perlinNoiseSameResultWithSameParameters`.